File: 0128-longest-consecutive-sequence/hashset/java/solution.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numsSet = new HashSet<>();
        for (int num : nums) {
            numsSet.add(num);
        }

        int maxLength = 0;
        for (int num : numsSet) {
            if (!numsSet.contains(num - 1)) { // num is start of a sequence
                int length = 1;
                while (numsSet.contains(num + length)) {
                    length++;
                }
                maxLength = Math.max(maxLength, length);
            }
        }
        return maxLength;
    }
}

File: 0128-longest-consecutive-sequence/hashset/cpp/solution.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> nums_set(nums.begin(), nums.end());
        int max_length = 0;
        for (int num : nums_set) {
            if (nums_set.find(num - 1) == nums_set.end()) { // num is start of sequence
                int length = 1;
                while (nums_set.find(num + length) != nums_set.end()) {
                    length++;
                }
                max_length = max(max_length, length);
            }
        }
        return max_length;
    }
};
